chemprop/data/utils.py

```diff
@@ -333,98 +373,166 @@ def get_data(path: str,
                     if loss_function == 'bounded_mse':
                         targets.append(float(value.strip('<>')))
                     else:
                         raise ValueError('Inequality found in target data. To use inequality targets (> or <), the regression loss function bounded_mse must be used.')
+                elif '[' in value or ']' in value:
+                    target = np.array(eval(value))
+                    if len(target.shape) == 1 and column in args.atom_targets:  # Atom targets saved as 1D list
+                        atom_targets.append(target)
+                        targets.append(target)
+                    elif len(target.shape) == 1 and column in args.bond_targets:  # Bond targets saved as 1D list
+                        bond_targets.append(target)
+                        targets.append(target)
+                    else:  # Bond targets saved as 2D list
+                        _is_mol = is_mol(smiles[0])
+                        keep_h = is_explicit_h(_is_mol)
+                        add_h = is_adding_hs(_is_mol)
+                        mol = make_mol(smiles[0], keep_h, add_h)
+                        bond_target_arranged = []
+                        for bond in mol.GetBonds():
+                            bond_target_arranged.append(target[bond.GetBeginAtom().GetIdx(), bond.GetEndAtom().GetIdx()])
+                        bond_targets.append(np.array(bond_target_arranged))
+                        targets.append(np.array(bond_target_arranged))
                 else:
                     targets.append(float(value))
+        elif extension in ['.pkl', '.pckl', '.pickle']:
+            _, row = row
+            row = row.to_dict()
 
-            # Check whether all targets are None and skip if so
-            if skip_none_targets and all(x is None for x in targets):
-                continue
+            smiles = [row[c] for c in smiles_columns]
 
-            all_smiles.append(smiles)
-            all_targets.append(targets)
+            # Atom targets
+            for a_target_name in args.atom_targets:
+                atom_target = np.array(row[a_target_name])
+                atom_targets.append(atom_target)
+
+            # Bond targets
+            _is_mol = is_mol(smiles[0])
+            keep_h = is_explicit_h(_is_mol)
+            add_h = is_adding_hs(_is_mol)
+            mol = make_mol(smiles[0], keep_h, add_h)
+            for b_target_name in args.bond_targets:
+                bond_target = np.array(row[b_target_name])
+                if len(bond_target.shape) == 1:
+                    bond_targets.append(bond_target)
+                else:
+                    bond_target_arranged = []
+                    for b in mol.GetBonds():
+                        bond_target_arranged.append(bond_target[b.GetBeginAtom().GetIdx(), b.GetEndAtom().GetIdx()])
+                    bond_targets.append(np.array(bond_target_arranged))
+            targets = atom_targets + bond_targets
```

Comment:
Are there any situations where the ordering of targets would deviate from this in a way that makes problems? (I don't think so but worth asking the question).

Reply:
If the bond information provided by users is a 2D array whose ordering of atoms should match the RDKit molecule object. In the meantime, the ordering of bonds determined in L449-450 also matches the RDKit molecule object. That is consistent so I think it is fine.